Source/Hiperion/Hiperion/App/scripts/controllers/user.js

```diff
@@ -34,51 +30,84 @@ angular.module('hiperionApp')
             }
         });
 
-        $scope.users = [];
-        $scope.name = "";
-        $scope.lastName = "";
-        $scope.age = "";
-
-        $scope.addNewUser = function() {
-            ngDialog.openConfirm({
-                template: 'App/views/userDialog.html',
-                className: 'ngdialog-theme-default',
-                preCloseCallback: 'preCloseCallbackOnScope',
-                scope: $scope
-            }).then(function(user) {
-                data.push(user);
-            }, function(reason) {
-            });
+        $scope.addNewUser = function () {
+            clearData();
+            openDialog();
         };
 
         $scope.editUser = function (user) {
-            user.$edit = true;
+            loadUserInDialog(user);
+            openDialog();
         };
 
-        $scope.saveUser = function (user) {
-            user.$edit = false;
+        $scope.saveUser = function () {
             var userdto = {
-                id: user.id,
-                name: user.name,
-                lastName: user.lastName,
-                age: user.age
+                id: $scope.id,
+                name: $scope.name,
+                lastName: $scope.lastName,
+                age: $scope.age
             };
 
-            $http.post('/api/User', JSON.stringify(userdto), { headers: { 'Content-Type': 'application/json' } });
+            $http.post('/api/User',
+                       JSON.stringify(userdto),
+                       { headers: { 'Content-Type': 'application/json' } })
+                 .success(function () {
+                     loadUsers();
+                 });
         };
 
-        $scope.cancelEditUser = function (user) {
-            user.$edit = false;
+        $scope.cancelEditUser = function () {
+            $scope.edit = false;
         };
 
         $scope.removeUser = function (user) {
             $http.delete('/api/User?id=' + user.id);
-            data = _.filter(data, function(item) {
+            data = _.filter(data, function (item) {

```

Comment:
We should reload the items from the backend, as we're doing with the other methods